Remarkable-Donkeys/bangazonWebApp
# Jf view categories

**The Name of this branch is jf-view-categories**


* **Please check if the PR fulfills these requirements**


* **What kind of change does this PR introduce?** (Bug Fix, Feature, Document Update...)
- Closer to finishing the View Categories page, just need to add links for the non-logged in Product Details pages for the individual products 
- Added an Authorize attribute to the Details method in the Products Controller

* **What Ticket is associated with this PR**
[Tichet #11](https://github.com/Remarkable-Donkeys/bangazonWebApp/issues/11)


* **What is the new behavior?**  (if this is a feature change)
User can now access the Categories via a link in the navigation bar


* **List all core functionality**
User can now access the Categories via a link in the navigation bar


* **Other information / Steps to Test**:

1. run the following
```
git fetch -all
git checkout jf-view-categories
```

2. delete all rows in the products table, yes, just do it

3. make sure you have two registered users in the AspNetUsers table with the usernames: jdoe@email.com & jsmith@email.com

4. run the application

5. click the Categories link in the navigation bar

6. should see a list of all the categories currently in the database with the top 3 products under each category

7. links for the products are not yet functional

8. make sure that when not logged in the following link redirects you to the login view, adjust the port number accordingly
[http://localhost:55646/Products/Details/26](http://localhost:55646/Products/Details/26)

***At least one person needs to approve pull request before merged to master. Indicate if you think more than one person should test the feature***
